### PR TITLE
Fixed many warnings.

### PR DESCRIPTION
**Pull request by MatthiasWM**

### CODE REVIEW

File: Emulator/PCMCIA/TPCMCIAController.h [hunk 1]
```diff
@@ -299,7 +299,7 @@ class TPCMCIAController
 										///< R/W for GetChipInfo.
 	KUInt32				mReg_3C00;		///< Value of xC003C00 (w   - ?? )
 	KUInt32				mReg_4000;		///< Value of xC004000 (w   - ?? )
-	KUInt32				mReg_4400;		//                     (    - ?? )
+//	KUInt32				mReg_4400;		//                     (    - ?? )
```
Comment (pguyot) [inline Emulator/PCMCIA/TPCMCIAController.h:302]: I would drop the line or fix the comment: the register is only read and always reads 0xFC.
Not sure what it means, though.

Comment (pguyot) [inline Emulator/PCMCIA/TPCMCIAController.h:302]: I would drop the line or fix the comment: the register is only read and always reads 0xFC.
Not sure what it means, though.

File: Emulator/Sound/TCoreAudioSoundManager.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@
 // Einstein.
 #include "Emulator/Log/TLog.h"
 
-#define MIN(a,b) ((a) < (b) ? (a) : (b))
+#define EINSTEIN_MIN(a,b) ((a) < (b) ? (a) : (b))
```
Comment (pguyot) [inline Emulator/Sound/TCoreAudioSoundManager.cpp:45]: If it's just used once, we could move it down/inline it.

File: Emulator/TNativePrimitives.cpp [hunk 3]
```diff
@@ -179,8 +179,6 @@ TNativePrimitives::SetEmulator( TEmulator* inEmulator )
 void
 TNativePrimitives::ExecuteNative( KUInt32 inInstruction )
 {
-	static KUInt16 n[0x1000] = { 0 };
```
Comment (pguyot) [inline Emulator/TNativePrimitives.cpp:182]: What was this used for? 🤔

Comment (MatthiasWM) [reply Emulator/TNativePrimitives.cpp:182]: I think it was used when I wrote boot progess into the screen overlay, which is no longer needed at current CPU speeds.

File: Emulator/PCMCIA/TPCMCIAController.h (hunk 1 above)
Comment (MatthiasWM) [reply Emulator/PCMCIA/TPCMCIAController.h:302]: I removed the variable, but I would still like to have its existance commented

File: Emulator/Sound/TCoreAudioSoundManager.cpp (hunk 2 above)
Comment (MatthiasWM) [reply Emulator/Sound/TCoreAudioSoundManager.cpp]: I changed it to std::min(). Easy. One other project in C++ rejects all std:: and template features. Stressful!

**Comment by MatthiasWM** [comment]
I fixed all warnings on Linux and most on macOS. I'd rather not touch iOS, and my Windows on ARM will not compile much, so I can't test that unfortunately.

Please feel free to comment. If I don't hear anything, I plan to Squash-n-Merge tomorrow by 12 CET.

**Pull request merged**